### PR TITLE
SEO-2443: Restricted Term Global Settings

### PR DESCRIPTION
**Pull request by JakeOTooleDev**
# [SEO-2443](https://jira.nyt.net/browse/SEO-2443): Restricted Term Global Settings

## Description

Added new UI and endpoints for adjusting restricted term settings at a global level.

## Motivation and Context

See JIRA Ticket [SEO-2443](https://jira.nyt.net/browse/SEO-2443).

## How Has This Been Tested?

Manual testing of new feature was performed.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/frontend-web/src/app/platform/restrictedTermsService.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// /services/simple/restrictedTerms
+export async function addRestrictedTerm(newTerm: any) {
+  const response: any = await axios.post(RESTRICTED_TERMS_URL, newTerm);
+  return response.data;
+}
+
+// /services/simple/restrictedTerms
+// Currently using type any while trying to figure out what the backend wants
+export async function updateRestrictedTerm(updatedTerm: any) {
+  console.log("updated term being sent", updatedTerm);
```
Comment (justinph) [inline packages/frontend-web/src/app/platform/restrictedTermsService.ts]: should prob nix the console.log

File: packages/frontend-web/src/app/scenes/Settings/Settings.tsx [hunk 2]
```diff
@@ -310,69 +305,61 @@ export class Settings extends React.Component<ISettingsProps, ISettingsState> {
...
+      : List([newValue]);
 
     this.setState({ preselects: updatedPreselects });
   }
 
   @autobind
   handleAddServiceUser(event: React.FormEvent<any>) {
     event.preventDefault();
-    console.log('got here');
+    console.log("got here");
```
Comment (justinph) [inline packages/frontend-web/src/app/scenes/Settings/Settings.tsx]: nix this?

File: packages/frontend-web/src/app/scenes/Settings/Settings.tsx [hunk 3]
```diff
@@ -1118,24 +1054,28 @@ export class Settings extends React.Component<ISettingsProps, ISettingsState> {
 
     return (
       <div {...css(STYLES.base)}>
-        <HeaderBar logout={logout} homeLink title="Settings"/>
+        <HeaderBar logout={logout} homeLink title="Settings" />
         <div {...css(STYLES.body)}>
           <h1 {...css(VISUALLY_HIDDEN)}>Open Source Moderator Settings</h1>
-          {this.renderUsers()}
+          {/* TEMP COMMENTING OUT TO MAKE DEV EASIER */}
+          {/* {this.renderUsers()} */}
```
Comment (justinph) [inline packages/frontend-web/src/app/scenes/Settings/Settings.tsx]: ?

File: packages/frontend-web/src/app/scenes/Settings/sections/AddRestrictedTerm.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  async handleAddNewTerm() {
+    try {
+      await globalRestrictedTerms.add({
+        term: this.state.newTerm,
+        score: parseFloat(this.state.newTermScore),
+      });
+      this.props.toggleDisplayAddTerm();
+      this.setState({ errorMessage: "" });
+    } catch (error) {
+      console.log("error occurred trying to add a new term", error.response);
```
Comment (justinph) [inline packages/frontend-web/src/app/scenes/Settings/sections/AddRestrictedTerm.tsx]: console.error?

File: packages/frontend-web/src/app/scenes/Settings/sections/AddRestrictedTerm.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+      this.setState({ errorMessage: error.response.data });
+    }
+    await this.props.getTerms();
+    console.log("After handle new term has an error");
```
Comment (justinph) [inline packages/frontend-web/src/app/scenes/Settings/sections/AddRestrictedTerm.tsx]: nix this?

File: packages/frontend-web/src/app/scenes/Settings/sections/RestrictedTerms.tsx [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+  }
+
+  @autobind
+  async getTerms() {
+    // get rid of Try/Catch when done with dev
+    try {
+      const terms = await globalRestrictedTerms.get();
+      this.setState({ terms: terms });
+    } catch (err) {
+      console.log("error getting terms", err);
```
Comment (justinph) [inline packages/frontend-web/src/app/scenes/Settings/sections/RestrictedTerms.tsx]: console.error?

**Review by justinph — COMMENTED** [review]
I dont see anything that scares me too much. lotta formatting changes i just glanced over. would like to see it in the browser to look at how some of the form changes work in the real world; hard to tell from just the code.

File: packages/frontend-web/src/app/scenes/Settings/sections/RestrictedTerms.tsx [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+    return (
+      // Almost more jaring with the menu disabled. Not sure what is best to do.
+      <div key="restrictedTermsSection" {...css(this.state.disableMenu && SETTINGS_STYLES.disableMenu)}>
+        <div key="heading" {...css(styles.heading)}>
+          <h2 {...css(styles.headingText)}>Restricted Terms</h2>
+        </div>
+        <div key="body" {...css(styles.section)}>
+          {/* Temporary wording. May need updating */}
+          <p>These terms are restricted and will cause a comment to be flagged.</p>
```
Comment (aalempijevic) [inline packages/frontend-web/src/app/scenes/Settings/sections/RestrictedTerms.tsx]: Maybe something more along the lines of 

`Comments containing these terms will receive Restricted Term tags with scoring corresponding to their assigned level`

**Comment by aalempijevic** [comment]
Overall looking good to me, I think just some of the console logs need to be cleaned up and then the settings should be moved to the bottom of the page as discussed.  I'll approve after that.

File: packages/frontend-web/src/app/scenes/Settings/Settings.tsx (hunk 3 above)
Comment (JakeOTooleDev) [reply packages/frontend-web/src/app/scenes/Settings/Settings.tsx]: WOOPS, need to uncomment this and move the Restricted Terms component way down.

**Review by aalempijevic — APPROVED** [review]

**Pull request merged**